lib/active_merchant/billing/gateways/redsys_rest.rb

```diff
@@ -378,6 +390,31 @@ def add_authentication(post, options)
         post[:DS_MERCHANT_MERCHANTCODE] = @options[:login]
       end
 
+      def add_stored_credentials(post, options)
+        return unless stored_credential = options[:stored_credential]
+
+        post[:DS_MERCHANT_COF_INI] = stored_credential[:initial_transaction] ? 'S' : 'N'
+
+        post[:DS_MERCHANT_COF_TYPE] = {
+          'recurring' => 'R',
+          'installment' => 'I'
+        }[stored_credential[:reason_type]] || 'C'
+
+        post[:DS_MERCHANT_IDENTIFIER] = 'REQUIRED' if stored_credential[:initiator] == 'cardholder'
+        post[:DS_MERCHANT_COF_TXNID] = stored_credential[:network_transaction_id] if stored_credential[:network_transaction_id].present?
+      end
+
+      def add_threeds_exemption_data(post, options)
+        return unless options[:three_ds_exemption_type]
+
+        if options[:three_ds_exemption_type] == 'moto'
+          post[:DS_MERCHANT_DIRECTPAYMENT] = 'MOTO'
```

Comment:
In order to ensure that this doesn't get overwritten in [add_direct_payment](https://github.com/activemerchant/active_merchant/pull/5132/files#diff-67c48e92bc50c99769a08c0c42d38a757185f1fc086944707387776d0e40d64cR292) we need to add `options[:three_ds_exemption_type] == 'moto'` as part of the `return if` in add_direct_payment

Reply:
Understood! Added!